numerics/swig/Numerics_typemaps_numericsmatrices.i

```diff
@@ -262,10 +262,18 @@
 
 
 #ifdef __cplusplus
-#define CSparseMatrix cs_sparse
+#ifdef SICONOS_USE_CXSPARSE
+#define CSparseMatrix cs_di_sparse
+#else
+#define CSparseMatrix cs_sparse;
+#endif
+#else
+#ifdef SICONOS_USE_CXSPARSE
+#define CSparseMatrix struct cs_di_sparse
 #else
```

Comment:
Yeah the CS_LONG stuff takes a little thought.  Actually we _do_ have control over it as far as I can tell.  That is, CXSparse exports both int and long versions (cs_di_sparse, cs_dl_sparse), and based on the value of CS_LONG that is set when Siconos is compiled, the cs_* functions will expand to the right version.  So SICONOS_INT64 seems to be the right way to control which CS_LONG we wish to use.  To interact with Siconos, the user must ensure he uses the same value of CS_LONG as was used when Siconos was compiled.  At least I think this is the case.  Hence I considered it okay to set in SparseMatrix.h based on the value of SICONOS_INT64 --- users of SparseMatrix.h _must_ use the value of CS_LONG that was specified during compilation, i.e., the value of SICONOS_INT64 in SiconosConfig.h.

Unless of course we want to make "di" and "dl" versions of all functions that take CSparseMatrix.

> Using CS_NAME(sparse) instead of hardcoding CSparseMatrix to a specific integer type would be nice.

I did try this actually, I was using a typedef to `CNAME(_sparse)` before, but swig didn't like it, gave syntax errors.

Reply:
ok that's sad that SWIG is in the way. For the CS_LONG and SICONOS_INT64 issue, I like the check. I was just worried about the user manually setting CS_LONG, or it being defined somewhere else. With a consistency check, it's all good.